### ISSUE TITLE
Evaluating model without tensorflow iterator

### ISSUE THREAD

**Issue by jeffbaena** [ppliuboy/SelFlow#27]
[EDIT I marked as solved as I found the issue, when loading the checkpoint the trailer '.index' should be removed.] Dear @ppliuboy, thanks for your great work. I would like to benchmark your model on some custom datasets.I have adapted the testing script from the Selflow repository, however:
- I cannot load the checkpoints provided, ('SelFlow/models/Sintel/supervise_finetune.index') the error I get is: 

>  2 root error(s) found.

>   (0) Not found: Tensor name "context_network/dilated_conv1/biases" not found in checkpoint files  > /home/user/optical_flow_networks/SelFlow/models/Sintel/supervise_finetune.index
	 [[node save/RestoreV2 (defined at tmp/test_imbalance_multiple_networks/test_selflow.py:98) ]]
	 [[save/RestoreV2/_3]]
  (1) Not found: Tensor name "context_network/dilated_conv1/biases" not found in checkpoint files /home/user/optical_flow_networks/SelFlow/models/Sintel/supervise_finetune.index
	 [[node save/RestoreV2 (defined at tmp/test_imbalance_multiple_networks/test_selflow.py:98) ]]
0 successful operations.
0 derived errors ignored.

- If I don't load the checkpoint the network works but of course I get nonsense outputs.

Is there an issue on how I load the model? Is there some preprocessing or post processing I should apply to the inputs or outputs? Attached you can find a snippet of my testing script and an output sample.

`from __future__ import division, print_function, absolute_import
import tensorflow as tf
import numpy as np
import os
import sys
import time
import cv2

from six.moves import xrange
from scipy import misc, io
from imageio import imwrite, imread
from tensorflow.contrib import slim

import matplotlib.pyplot as plt
from networks.SelFlow.network import pyramid_processing
from networks.SelFlow.datasets import BasicDataset
from networks.SelFlow.utils import average_gradients, lrelu, occlusion, rgb_bgr,mvn
from networks.SelFlow.data_augmentation import flow_resize
from networks.SelFlow.flowlib import flow_to_color, write_flo
from networks.SelFlow.warp import tf_warp
from utils.flow import flow_to_png_middlebury
os.environ["CUDA_VISIBLE_DEVICES"]="0"

## Input
sintel_pth = '/home/ssavian/training/sintel/training/clean/alley_1'
im1_pth = os.path.join(sintel_pth,"frame_0001.png")
im2_pth = os.path.join(sintel_pth,"frame_0002.png")
im3_pth = os.path.join(sintel_pth,"frame_0003.png")
gnd_pth = '/home/user/training/sintel/training/flow/alley_1/frame_0002.flo'
gnd_vis = cv2.readOpticalFlow(gnd_pth)
output_png_pth = '/home/user/training/plots/test_gnd_selflow.png'
imwrite(output_png_pth, flow_to_png_middlebury(np.transpose(gnd_vis, (2, 0, 1))))
##checkpoint
checkpoint_path = ''
model_ckpt = ''

model_checkpoint_path = os.path.join(checkpoint_path,model_ckpt)
##

img0 = tf.image.decode_png(tf.read_file(im1_pth), channels=3)
img0 = tf.cast(img0, tf.float32)
img1 = tf.image.decode_png(tf.read_file(im2_pth), channels=3)
img1 = tf.cast(img1, tf.float32)
img2 = tf.image.decode_png(tf.read_file(im3_pth), channels=3)
img2 = tf.cast(img2, tf.float32)

img0 = img0 / 255.
img1 = img1 / 255.
img2 = img2 / 255.
is_normalize_img =True
if is_normalize_img:
    img0 = mvn(img0)
    img1 = mvn(img1)
    img2 = mvn(img2)

img_shape = tf.shape(img1)
h = img_shape[0] 
w = img_shape[1]

img0 = tf.expand_dims(img0, 0)
img1 = tf.expand_dims(img1, 0)
img2 = tf.expand_dims(img2, 0)

new_h = tf.where(tf.equal(tf.mod(h, 64), 0), h, (tf.to_int32(tf.floor(h / 64) + 1)) * 64)
new_w = tf.where(tf.equal(tf.mod(w, 64), 0), w, (tf.to_int32(tf.floor(w / 64) + 1)) * 64)

batch_img0 = tf.image.resize_images(img0, [new_h, new_w], method=1, align_corners=True)
batch_img1 = tf.image.resize_images(img1, [new_h, new_w], method=1, align_corners=True)
batch_img2 = tf.image.resize_images(img2, [new_h, new_w], method=1, align_corners=True)

flow_fw, flow_bw = pyramid_processing(batch_img0, batch_img1, batch_img2, train=False, trainable=False,
                                      is_scale=True)
flow_fw['full_res'] = flow_resize(flow_fw['full_res'], [h, w], method=1)
flow_bw['full_res'] = flow_resize(flow_bw['full_res'], [h, w], method=1)

flow_fw_color = flow_to_color(flow_fw['full_res'], mask=None, max_flow=256)
flow_bw_color = flow_to_color(flow_bw['full_res'], mask=None, max_flow=256)

restore_vars = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES)
saver = tf.train.Saver(var_list=restore_vars)
sess = tf.Session()
sess.run(tf.global_variables_initializer())

model_checkpoint_path = '/home/user/optical_flow_networks/SelFlow/models/Sintel/supervise_finetune.index'
saver.restore(sess, model_checkpoint_path)
output = flow_fw['full_res'].eval(session=sess)
output_color = flow_fw_color.eval(session=sess)

temp = output[0,:,:,:].transpose([2, 0, 1])
imwrite('/home/user/training/plots/test_selflow_ouput.png', flow_to_png_middlebury(temp))`

![test_selflow_ouput](https://user-images.githubusercontent.com/18122010/115429905-289ed980-a204-11eb-9dc4-734c1cb7423b.png)
![test_gnd_selflow](https://user-images.githubusercontent.com/18122010/115429919-2b99ca00-a204-11eb-9b1a-3a9d4c96809e.png)
